themes/default/content/challenge/holiday-shopping.md

```diff
@@ -0,0 +1,269 @@
+)
+
+# AWS export
+pulumi.export("app url", Output.concat("http://", lb.load_balancer.dns_name))
+```
+
+Run `pulumi up` and select yes to perform the update. Pulumi will output the `app url`. Go to the URL, and you will be presented with Pulumipus' Grocery List web application. You can try to add some items, but nothing will happen because there is no database connected yet. You will do that in the next step.
+
+#### Step 3. Create a MongoDB Atlas database
+
+The web app stores the grocery list items inside a MongoDB database. MongoDB Atlas is a fully managed cloud database that you will be using for this application. Add the below before the code from the previous steps in `__main__.py`. This code creates a MongoDB project, sets up the network access controls, creates a Free Tier cluster in `US_WEST_2` in AWS, and a database user that is scoped to the created cluster.
+
+```python
+# MongoDB Atlas configs
+db_username = config.get("dbUser", "test-username")
+db_password = config.get_secret_object("dbPassword", "test-password")
+atlas_org_id = config.get("orgID")
+
+# Create MongoDB Project
+mongo_project = mongodb.Project("mongo_project", org_id=atlas_org_id)
+
+# Open access to all IPs
+mongo_acl = mongodb.ProjectIpAccessList("mongo_acl",
+    cidr_block="0.0.0.0/0",
+    comment="Open access for backend",
+    project_id=mongo_project.id,)
+
+# Create Free Tier cluster
+mongo_cluster = mongodb.Cluster("mongo-cluster",
+    backing_provider_name="AWS",
+    project_id=mongo_project.id,
+    provider_instance_size_name="M0",
+    provider_name="TENANT",
+    provider_region_name="US_WEST_2")
+
+# Create Database user and give access to the cluster and database
+mongo_user = mongodb.DatabaseUser("db_user",
+    auth_database_name="admin",
+    labels=[mongodb.DatabaseUserLabelArgs(
+        key="project",
+        value="pulumi",
+    )],
+    password=db_password,
+    project_id=mongo_project.id,
+    roles=[
+        mongodb.DatabaseUserRoleArgs(
+            database_name="grocery-list",
+            role_name="readWrite",
+        ),
+        mongodb.DatabaseUserRoleArgs(
+            database_name="admin",
+            role_name="readAnyDatabase",
+        ),
+    ],
+    scopes=[
+        mongodb.DatabaseUserScopeArgs(
+            # Extracts the cluster name to add to database scopes
+            name=Output.all(mongo_cluster.srv_address).apply(lambda v: re.split("\.|\/\/", v[0])[1]),
+            type="CLUSTER",
+        ),
+    ],
+    username=db_username
+)
+
+# MongoDB Atlas export
+pulumi.export("mongo connection string",
+    Output.format("mongodb+srv://{0}:{1}@{2}", db_username, db_password,
+        Output.all(mongo_cluster.srv_address).apply(lambda v: v[0].split("//"))[1])
+)
+```
+
+Next, you need to set the config values for your MongoDB Cloud org, database username, and database password.
+
+```shell
+pulumi config set orgID [value]
+pulumi config set dbUser [value]
+pulumi config set dbPassword [value] --secret
```

Comment:
If it's possible to link directly to the locations for these values in Atlas, we should do that just to keep folks from getting lost. Should also mention that the user and pass will be used to provision, so they don't need to be obtained from Atlas first.

I also remember getting the creds set up took a bit of figuring out -- we should add some instructions for this also (like what perms to use, where to get the keys from and how to set them, etc.).

Reply:
I can't link directly to it, but I did mention where to find it